lib/Doctrine/Common/Cache/CacheProvider.php

```diff
@@ -286,6 +297,26 @@ protected function doSaveMultiple(array $keysAndValues, $lifetime = 0)
     abstract protected function doSave($id, $data, $lifeTime = 0);
 
     /**
+     * Default implementation of doDeleteMultiple. Each driver that supports multi-delete should override it.
+     *
+     * @param array $keys Array of keys to delete from the cache
+     *
+     * @return bool TRUE if the operation was successful, FALSE if it wasn't.
+     */
+    protected function doDeleteMultiple(array $keys)
+    {
+        $success = true;
+
+        foreach ($keys as $key) {
+            if (!$this->doDelete($key)) {
+                $success = false;
```

Comment:
I know that this may sound like an ugly question, but is this scenario covered by the tests? (doesn't seem like it)